elixir/apps/web/lib/web/live/settings/dns.ex

```diff
@@ -38,35 +38,53 @@ defmodule Web.Settings.DNS do
       </:action>
 
       <:help>
-        Configure the upstream resolver used by connected Clients.
-        Queries for Resources will <strong>always</strong> use Firezone's internal DNS.
-        All other queries will use the resolvers configured here or the Client's
-        system resolvers if none are configured.
+        <p>
+          Configure the search domain and upstream resolvers used by devices when the Firezone Client is connected.
+        </p>
       </:help>
 
       <:content>
         <div class="max-w-2xl px-4 py-8 mx-auto">
-          <h2 class="mb-4 text-xl text-neutral-900">Client DNS</h2>
+          <.form for={@form} phx-submit={:submit} phx-change={:change}>
+            <.flash kind={:success} flash={@flash} phx-click="lv:clear-flash" />
 
-          <.flash kind={:success} flash={@flash} phx-click="lv:clear-flash" />
+            <.inputs_for :let={config} field={@form[:config]}>
+              <h2 class="mb-4 text-xl text-neutral-900">Search Domain</h2>
 
-          <% empty? =
-            Domain.Repo.Changeset.empty?(@form.source) and
-              Enum.empty?(@account.config.clients_upstream_dns) %>
+              <p class="mb-4 text-neutral-500">
+                The search domain, or default DNS suffix, will be appended to all single-label DNS queries made by Client devices
+                while connected to Firezone.
+              </p>
 
-          <p :if={not empty?} class="mb-4 text-neutral-500">
-            Upstream resolvers will be used by Clients in the order they are listed below.
-          </p>
+              <div class="mb-12">
+                <.input field={config[:search_domain]} placeholder="E.g. example.com" />
+                <p class="mt-2 text-sm text-neutral-500">
+                  Enter a valid FQDN to append to single-label DNS queries. The
```

Comment:
Is there any chance a customer might want to use a top level domain as their search domain?  As an example, what about if a customer runs their own resolver and uses something like `.lan` for their internal domain since it's not a valid TLD?

Reply:
I haven't come across any request for this yet - typically these are used in corp intranet style access scenarios where the FQDN itself is valid.

That said, I can't guarantee we won't be asked to support it, but if so, better to err on the conservative side for now I think.

Reply:
Totally fair.

So we will just take a stand that we won't support my example use case.